"""2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля в качестве делителя программа
должна корректно обработать эту ситуацию и не завершиться с ошибкой."""


class MyError(Exception):
    def __init__(self, num):
        self.num = num

num = input("Введите число: ")
num_1 = 100
try:
    num = int(num)
    if num == 0:
        raise MyError("Деление на ноль недопустимо!")
except ValueError:
    print("Вы ввели не число")
except MyError as err:
    print(err)
else:
    print(f"Все хорошо. Ваше число: {num}")
    print(f"Результат: {num_1 / num}")
finally:
    print("Программа завершена!")
